controllers/preflightvalidation_controller.go

```diff
@@ -0,0 +1,239 @@
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	kmmv1beta1 "github.com/qbarrand/oot-operator/api/v1beta1"
+	"github.com/qbarrand/oot-operator/internal/filter"
+	"github.com/qbarrand/oot-operator/internal/preflight"
+	"github.com/qbarrand/oot-operator/internal/statusupdater"
+	"github.com/qbarrand/oot-operator/internal/utils"
+)
+
+const reconcileRequeueInSeconds = 60
+
+// ClusterPreflightReconciler reconciles a PreflightValidation object
+type PreflightValidationReconciler struct {
+	client        client.Client
+	filter        *filter.Filter
+	statusUpdater statusupdater.PreflightStatusUpdater
+	preflight     preflight.PreflightAPI
+}
+
+func NewPreflightReconciler(
+	client client.Client,
+	filter *filter.Filter,
+	statusUpdater statusupdater.PreflightStatusUpdater,
+	preflight preflight.PreflightAPI) *PreflightValidationReconciler {
+	return &PreflightValidationReconciler{
+		client:        client,
+		filter:        filter,
+		statusUpdater: statusUpdater,
+		preflight:     preflight}
+}
+
+func (r *PreflightValidationReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("preflightvalidation").
+		For(&kmmv1beta1.PreflightValidation{}).
+		Watches(
+			&source.Kind{Type: &kmmv1beta1.Module{}},
+			handler.EnqueueRequestsFromMapFunc(r.filter.FindPreflightsForModule),
+			builder.WithPredicates(r.filter.PreflightReconcilerModulePredicate()),
+		).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: 1,
+		}).
+		Complete(r)
+}
+
+//+kubebuilder:rbac:groups=kmm.sigs.k8s.io,resources=modules,verbs=get;list;watch
+//+kubebuilder:rbac:groups=kmm.sigs.k8s.io,resources=preflightvalidations,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=kmm.sigs.k8s.io,resources=preflightvalidations/status,verbs=get;update;patch
+
+// Reconcile Reconiliation entry point
+func (r *PreflightValidationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+
+	reconRes := ctrl.Result{}
+
+	log := ctrl.LoggerFrom(ctx)
+	log.Info("Start PreflightValidation Reconciliation")
+
+	pv := kmmv1beta1.PreflightValidation{}
+	err := r.client.Get(ctx, types.NamespacedName{Namespace: req.Namespace, Name: req.Name}, &pv)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("PreflightValidation reconcile success. Reconciliation object not found, probably deleted. Not reconciling")
+			return reconRes, nil
+		} else {
+			log.Error(err, "preflight validation reconcile failed to find object")
+			return reconRes, err
+		}
+	}
+
+	reconCompleted, err := r.runPreflightValidation(ctx, &pv)
+	if err != nil {
+		log.Error(err, "runPreflightValidation failed")
+		return reconRes, err
+	}
+
+	if reconCompleted {
+		log.Info("PreflightValidation reconciliation success")
+		return ctrl.Result{}, nil
+	}
+	log.Info("PreflightValidation reconciliation requeue")
+	return ctrl.Result{RequeueAfter: time.Second * reconcileRequeueInSeconds}, nil
```

Comment:
we want to reschedule the reconciliation, so that user might fix the validation. We cannot count on Module update, since the correction might include only pre-building an image

Reply:
What is the default requeue time if we only use `ctrl.Result{Requeue: true}`?

Reply:
we want to control the timeout. In our case we don't want the re queuing to happen quickly

Reply:
I agree, just wondering what is the default time.

Reply:
Why not use the default backoff scheme?

Reply:
i don't see the point to use it. backoff scheme is used in case of an errors, in our case we just want to give a user a specific amount of time to fix something from the beginning. The current timeout value will be for sure changed once we have customer using this feature

Reply:
OK then.